static/js/settings_linkifiers.js

```diff
@@ -109,6 +128,59 @@ export function build_page() {
         });
     });
 
+    $(".admin_filters_table").on("click", ".edit", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const btn = $(this);
+        const filter_id = Number.parseInt(btn.attr("data-filter-id"), 10);
+        const modal = open_linkifier_edit_form(filter_id).modal;
+
+        modal.find(".submit_linkifier_info_change").on("click", (e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            const change_filter_button = $(".submit_linkifier_info_change");
+            change_filter_button.prop("disabled", true);
+
+            const url = "/json/realm/filters/" + encodeURIComponent(filter_id);
+            const pattern = modal.find("#edit_filter_pattern").val().trim();
+            const url_format_string = modal.find("#edit_filter_url_format_string").val().trim();
+            const data = {pattern, url_format_string};
+            const filter_status = modal.find(".edit-filter-status").expectOne();
+            const pattern_status = modal.find("#edit-filter-pattern-status").expectOne();
+            const format_status = modal.find("#edit-filter-format-status").expectOne();
+            const opts = {
+                success_continuation() {
+                    change_filter_button.prop("disabled", false);
+                    overlays.close_modal("#linkifier-info-form-modal");
+                },
+                error_continuation(xhr) {
+                    change_filter_button.prop("disabled", false);
```

Comment:
Maybe worth adding a brief comment here that the endpoint uses the somewhat uncommon in our codebase Django ValidationError system that returns somewhat complicated error dictionaries.

This isn't a pattern we want to use in general, but it's probably OK to have it here given that the other linkifier endpoints have it too.

That said, it'd be nice to have a prep commit that extracts the code here that's common with the "add linkifier" code path as a function that we can call from both places rather than copying it.

In general, we try to avoid copying substantial blocks of code, as it tends to result in someone later fixing bugs in only one of the two places.

Reply:
Done.